File: src/main/java/org/vors/pairbot/model/Participant.java
package org.vors.pairbot.model;

import javax.persistence.*;

@Entity
@Table(name = "participants")
@IdClass(ParticipantId.class)
public class Participant {

    @Id
    @ManyToOne
    @JoinColumn(name = "user_pk", referencedColumnName = "pk")
    private UserInfo user;

    @Id
    @ManyToOne
    @JoinColumn(name = "event_pk", referencedColumnName = "pk")
    private Event event;

    @Column
    private Boolean accepted;

    @Column
    private Boolean host = false;

    public Participant(){}

    public Participant(UserInfo user){
        this.user = user;
    }

    public UserInfo getUser() {
        return user;
    }

    public void setUser(UserInfo user) {
        this.user = user;
    }

    public Event getEvent() {
        return event;
    }

    public void setEvent(Event event) {
        this.event = event;
    }

    public Boolean isAccepted() {
        return accepted;
    }

    public void setAccepted(Boolean accepted) {
        this.accepted = accepted;
    }

    public Boolean getHost() {
        return host;
    }

    public void setHost(Boolean host) {
        this.host = host;
    }
}

File: src/main/java/org/vors/pairbot/model/Participant.kt
package org.vors.pairbot.model

import javax.persistence.*

@Entity
@Table(name = "participants")
@IdClass(ParticipantId::class)
class Participant(
        @Id @ManyToOne @JoinColumn (name = "user_pk", referencedColumnName = "pk")
        var user: UserInfo,
        @Id @ManyToOne @JoinColumn(name = "event_pk", referencedColumnName = "pk")
        var event: Event,
        var status: ParticipantStatus = ParticipantStatus.NO_RESPONSE,
        var host: Boolean = true
) {
        override fun equals(other: Any?): Boolean {
                if (this === other) return true
                if (javaClass != other?.javaClass) return false

                other as Participant

                if (user != other.user) return false
                if (event != other.event) return false

                return true
        }

        override fun hashCode(): Int {
                var result = user.hashCode()
                result = 31 * result + event.hashCode()
                return result
        }
}
